monkey/monkey_island/cc/models/agent_signals.py

```diff
@@ -0,0 +1,8 @@
+from datetime import datetime
+from typing import Optional
+
+from common.base_models import InfectionMonkeyBaseModel
+
+
+class AgentSignals(InfectionMonkeyBaseModel):
+    terminate: Optional[datetime]
```

Comment:
This needs some kind of identifier for the agent too. The agent ID or a nested `Agent` pydantic object?

Reply:
It doesn't need it because this data isn't stored anywhere ATM. It's generated on the fly when it is requested.

Reply:
Got it. `AgentSignalsService.get_signals()` calls `agent_terminated()` (which accepts an `Agent` object) and creates an `AgentSignals()` object then.